core/controllers/base.py

```diff
@@ -148,7 +148,7 @@ class BaseHandler(webapp2.RequestHandler):
 
     URL_PATH_ARGS_SCHEMAS = None
     HANDLER_ARGS_SCHEMAS = None
-    ARGS_WHICH_DO_NOT_NEED_SCHEMA_VALIDATION = ['csrf_token']
+    ARGS_WHICH_DO_NOT_NEED_SCHEMA_VALIDATION = ['csrf_token', 'source']
```

Comment:
I get why csrf_token is excluded ... but why "source"? Where are we using it? I don't think it should be considered as "special".

Reply:
Sources are basically parent url, like if we are on the ```/create/``` handler and we make a request to ```/createhandler/rights/``` then first url is acting as a parent url.
In the wiki page I have showed the example of all the arguments and their values. [here](https://github.com/oppia/oppia/wiki/Writing-schema-for-handler-args#debugging-tricks)


------------------------------------
(u'Request url = ', 'http://localhost:8181/createhandler/rights/QuWbhgRTovXr')
(u'Handler class name = ', 'ExplorationRightsHandler')
(u'Arguments = ', ['csrf_token', 'payload', 'source'])
Iterating over arguments...
('csrf_token', u'1622997677/wOV5q43bIZf1cvOhCB4vrQ==')
('payload', u'{"version":1,"new_member_role":"owner","new_member_username":"nikhil"}')
('source', u'http://localhost:8181/create/QuWbhgRTovXr#/settings')
------------------------------------

Reply:
Oh, I see. Could you point me to the location in the code where it gets set, please?

Reply:
(I want to understand whether this is something related to the webapp2 framework, or if it is something that we control. I'm a bit confused why we need to pass it, given that we can just look at self.request.uri (right?)

Reply:
It is feature of webapp2, and this arg is only present when we have a parent url.
 ```request.uri()``` help us to get the url from which the handler is requested not the parent url.
And since the source contains the url we do not have to write schema for it.

Looking forward for youe suggestions.

Reply:
Oh, I see, got it. OK, sounds fine then, please leave a comment above this constant explaining clearly what each arg is and why it is excluded. Link to authoritative references/documentation as needed. That way, we don't have people just adding to that list any time they have an arg name that happens to be a bit hard to validate.

Reply:
Done, please take a look.

Reply:
The docstring is good, but doesn't fully address the issue.

Specifically, for the "source" docstring, you should state explicitly that the 'source' is provided by webapp2 and is only present when we have a parent URL (and give the example of /create/ making a request to /createhandler/rights). Please also provide a link to the relevant webapp2 documentation describing it.

Reply:
@seanlip I have updated the docstring but I cannot provide any refernce link since I have not found any relevant webapp2 documentation.

Reply:
@seanlip 'source' is added by us in the file request-interceptor.service.ts from [here](https://github.com/oppia/oppia/blob/bd000e843d00e6b8ced66c23ba2df475bca18fd1/core/templates/services/request-interceptor.service.ts#L97).

Reply:
You still haven't explained this in the comment. You've just described what the args contain but haven't provided reasoning for why these args are special and don't need to be validated by the schema validation architecture...

Please provide the missing details and justify why those args should be included in this list.

Reply:
Done, please take a look.

Reply:
Still doesn't seem correct.

The reason we do not validate such args is that they are not in the payload .... right? They are siblings of the payload and the schema validation architecture is intended to validate only the payload. Isn't that the case?

Please discuss this with your mentor before you make further changes so that he can confirm whether what you are doing makes sense.

Reply:
Updated please take a look.